connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -181,6 +185,17 @@ message ExecutePlanResponse {
       string metric_type = 3;
     }
   }
+
+  // This message is only used by dataframe.observe if the observation is Observation.
+  message ObservedMetrics {
+
+    repeated ObservedMetricsObject metrics_objects = 1;
+
+    message ObservedMetricsObject {
```

Comment:
This name is a bit awkward. If we go with the repeated approach, see the comment above, the just name this `ObservedMetrics`.